---
title: "STA426 Week 5 Exercise"
author: "Franziska Lampart (franziska.lampart@uzh.ch)"
date: "October 22, 2016"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(biomaRt)
```

## Part 1
### Extracting Information from Ensembl
Using the biomaRT library for R, a small set of annotations was extracted from the *Gallus gallus* genome from the BioMart database "Ensemble Genes".

```{r}
ensembl = useMart("ensembl")
ensembl = useDataset("ggallus_gene_ensembl", mart=ensembl)

Ggallus_annot <- getBM(attributes=c('ensembl_gene_id','transcript_count',
                                    'percentage_gc_content'), mart = ensembl)
Ggallus_trancripts <- getBM(attributes=c('ensembl_transcript_id',
                                         'transcript_length'), mart = ensembl)
```
With the arguments attributes the annotations of interest were filtered from the data set. There are `r nrow(listAttributes(ensembl))` attributes available to search for. In the present exercise the attributes for the *Gene_ID*, *Transcript_Count*, *GC_Content*, *Transcript_Length* and *Transcript_ID* where used.

## Analysing and Plotting the data
### Number of Genes and Transcripts
There are `r nrow(Ggallus_annot)` genes annotated for *Gallus gallus* with a total number of `r sum(Ggallus_annot[,2])` transcripts.
The number of transcripts per gene is visualized below;

```{r, echo=FALSE}
hist(Ggallus_annot[,2], breaks=10, xlab = "Trancripts per Gene",
     main = "Frequency of transcripts per gene in Gallus gallus",
     col = "grey", labels = TRUE)
```

### Distribution of Transcript length
The distribution of transcript length is based on the transcripts including the 5'- and 3'-UTR, with the shortest transcript length of `r min(Ggallus_trancripts[,2])` and a maximum length of `r max(Ggallus_trancripts[,2])`. The average transcript length is `r round(mean(Ggallus_trancripts[,2]))`.

```{r, echo=FALSE}
plot(density(Ggallus_trancripts[,2]), main = "Distribution of transcript Length in 
     Gallus gallus", xlab = "Transcript length", col = "red")
```

### GC content of the genes
Below the density distribution of the GC content of all annotated genes is shown.
The average GC content is `r round(mean(Ggallus_annot[,3]), 2)`

```{r, echo=FALSE}
plot(density(Ggallus_annot[,3]), main = "Distribution of GC content in Gallus gallus", 
     xlab = "GC content in %", col = "blue")
```

With this library it is possible to quickly extract different annotation information of variable organisms, which facilitates the comparison.
As an example comparing the GC content of three different species.

```{r, echo=FALSE}
ensemblD = useDataset("dmelanogaster_gene_ensembl", mart=ensembl)
ensemblS = useDataset("scerevisiae_gene_ensembl", mart=ensembl)

Dmelanog_annot <- getBM(attributes=c('ensembl_gene_id', 'percentage_gc_content'),
                       mart = ensemblD)
Scerevis_annot <- getBM(attributes=c('ensembl_gene_id', 'percentage_gc_content'),
                        mart = ensemblS)
boxplot(Ggallus_annot[,3], Dmelanog_annot[,2], Scerevis_annot[,2],
        names = c("G. gallus", "D. melanogaster", "S. cerevisiae"),
        main = "GC content of genes from differen organisms", ylap = "%GC")
```

##Appendix

###Session Info
```{r echo = FALSE}
sessionInfo()
```
